doc/array_and_matrix_manipulation.md

```diff
@@ -0,0 +1,271 @@
+% Array and Matrix Manipulation
+
+ArrayFire provides several different methods for manipulating arrays and matrices.
+The functionality includes:
+
+* [moddims()](#moddims) - change the dimensions of an array without changing the data
+* [flat()](#flat) - flatten an array to one dimension
+* [flip()](#flip) - flip an array along a dimension
+* [join()](#join) - join up to 4 arrays
+* [reorder()](#reorder) - changes the dimension order within the array
+* [shift()](#shift) - shifts data along a dimension
+* [tile()](#tile) - repeats an array along a dimension
+* [transpose()](#transpose) - performs a matrix transpose
+
+Below we provide several examples of these functions and their use.
+
+### moddims()
```

Comment:
Yea, good point. Probably helpful as a reference for people getting started (before they have their setup/etc)